### ISSUE TITLE
Parameters of ScriptBlock in Register-ArgumentComplete are set incorrectly

### ISSUE THREAD

**Issue by ar-std** [PowerShell/PowerShell#24031]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I want to create an argument completer by using `Register-ArgumentCompleter`. If setting up the ArgumentCompleter as

```pwsh
$s = {
    param($commandName, $parameterName, $wordToComplete, $commandAst,
          $fakeBoundParameters )

    @("commandName:         $commandName",
      "parameterName:       $parameterName",
      "wordToComplete:      $wordToComplete",
      "commandAst:          $commandAst",
      "fakeBoundParameters: $fakeBoundParameters") | Set-Content log.txt
}
Register-ArgumentCompleter -CommandName test -ScriptBlock $s
```

The parameters of the script block are populated incorrectly. It is populated as if the `-Native` option is used.

### Expected behavior

```console
PS> test abc<TAB>
PS> Get-Content .\log.txt
commandName:         test
parameterName:       
wordToComplete:      abc
commandAst:          test abc
fakeBoundParameters:
```

### Actual behavior

```console
PS> test abc<TAB>
PS> Get-Content .\log.txt
commandName:         abc     
parameterName:       test abc
wordToComplete:      8       
commandAst:                  
fakeBoundParameters:
```

### Environment data

```powershell
PS> $PSVersionTable 

Name                           Value
----                           -----
PSVersion                      7.4.3
PSEdition                      Core
GitCommitId                    7.4.3
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}       
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by jhoneill** [comment]
You need to specify which parameter you want to complete in register-ArgumentCompleter. 
Then the parameter name will be completed as well. 
![image](https://github.com/PowerShell/PowerShell/assets/16209593/ab5cbf18-fe62-4ee7-bbf8-402c3c34bcd8)

The above has tab completed from nothing, but with ABC [tab] you'll get ABC in `WordToComplete`

**Comment by mklement0** [comment]
It does appear that if you don't specify a `-ParameterName` argument, the `-Native` switch is _implied_ (which is an unfortunate parameter design).

With `-Native` in effect, the parameter list changes to `param($wordToComplete, $commandAst, $cursorColumn)`.

**Comment by ar-std** [comment]
Sorry Guys—my bad! I got confused by the documentation's description of `-Native`. The example using `-Native` uses a native command to generate the argument completion for a native command and I thought I need `-Native` for the argument completer and not for the native command I need completion for. Given that my `test` command is native I should specify `-Native`. 

Nonetheless, applying `-Native` implicitly isn't a good habit as it effects the script block's parameters. I'd rather like to get an error message like "Unable to determine parameter name for 'test'. If 'test' is a native command, consider using the option '-Native' and change the parameters of your script block." when calling `Register-ArgumentCompleter`

**Comment by mklement0** [comment]
I agree that the parameter design is unfortunate and that it would have been better to make `-ParameterName` mandatory in the absence of `-Native` (in which case you'd get prompted for a `-ParameterName` value in the absence of both, in interactive scenarios).

However, changing this now could break existing code, so I don't think such a change would be considered.

Instead, I suggest improving the docs:
* https://github.com/MicrosoftDocs/PowerShell-Docs/issues/11248

**Comment by mklement0** [comment]
@ar-std, now that the docs have been updated, I suggest closing this issue.

(Your suggestion at https://github.com/MicrosoftDocs/PowerShell-Docs/issues/11248#issuecomment-2220924531 would again amount to a breaking change, so I don't think it will be entertained.)
